### PR TITLE
Switch to typescript and add Dark Mode

### PR DESCRIPTION
**Pull request by maxslarsson**

### CODE REVIEW

**Comment by maxslarsson** [comment]
* **#56** <a href="https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/56?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maxslarsson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by ZachGarcia42 — CHANGES_REQUESTED** [review]
Can you add the Dockerfile back so I can pull that image too?

**Review by ZachGarcia42 — COMMENTED** [review]
I checked out your code, and grpc works fine for me. We should make sure it works for you too though. Just to double check, here are the steps I took:

1. navigate to ground-server
2. run npm install
3. run npm run dev
4. docker run -it -d --network=host ghcr.io/cornellrocketryteam/fill-station
5. Navigate to localhost:3000/grpc and refresh if necessary

File: ground-server/package-lock.json [hunk 1]
```diff
@@ -64,7 +73,6 @@
       "resolved": "https://registry.npmjs.org/@eslint/eslintrc/-/eslintrc-2.1.4.tgz",
       "integrity": "sha512-269Z39MS6wVJtsoUl10L60WdkhJVdPG24Q4eZTH3nnF6lpvSShEK3wQjDX9JRWAUPvPh7COouPpU9IrqaZFvtQ==",
       "dev": true,
-      "license": "MIT",
```
Comment (ZachGarcia42) [inline ground-server/package-lock.json:67]: Very unimportant, but do you know why the MIT licenses were removed here?

**Comment by maxslarsson** [comment]
I get the error:

Error: Cannot Connect. Is the Fill Station Running? Error: 14 UNAVAILABLE: No connection established. Last error: connect ECONNREFUSED 127.0.0.1:50051 (2024-09-26T02:43:08.025Z)

**Comment by ZachGarcia42** [comment]
> I get the error:
> 
> Error: Cannot Connect. Is the Fill Station Running? Error: 14 UNAVAILABLE: No connection established. Last error: connect ECONNREFUSED 127.0.0.1:50051 (2024-09-26T02:43:08.025Z)

I'll investigate on a Pi tomorrow to make sure it's not an ARM issue. Let's hold off merging until then. If that works for me, it's probably some firewall rule on your Mac?

**Comment by maxslarsson** [comment]
Yes probably, sounds good

File: ground-server/src/app/grpc/commandClient.js [hunk 2]
```diff
@@ -1,41 +1,43 @@
 "use server";
+
 var PROTO_PATH = __dirname + '/../../../../../protos/command.proto';
 var grpc = require('@grpc/grpc-js');
```
Comment (ZachGarcia42) [inline ground-server/src/app/grpc/commandClient.js:4]: Change to 

import * as grpc from '@grpc/grpc-js';
import * as protoLoader from '@grpc/proto-loader';

for ESLint

File: ground-server/package-lock.json (hunk 1 above)
Comment (maxslarsson) [reply ground-server/package-lock.json:67]: I don't know, but this file is auto generated by npm. Maybe eslint removed that themselves?

**Review by ZachGarcia42 — APPROVED** [review]
Looks good! Thanks for the work!

**Comment by maxslarsson** [comment]
### Merge activity

* **Sep 26, 5:03 PM EDT**: @maxslarsson started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/56).

**Pull request merged**